### DISCUSSION TITLE
RPM repo with pgexporter - Rocky 8

### DISCUSSION THREAD

**Discussion by caleno** [pgexporter/pgexporter#109] [General]
Hi, quick question

According to [doc/tutorial/02_custom_metrics.md](https://github.com/pgexporter/pgexporter/blob/main/doc/tutorial/01_install.md#install-pgexporter) there is a  pgexporter RPM package available in the [PostgreSQL YUM repository](https://yum.postgresql.org/) 

I can only find the pgexporter_ext package. I have misunderstood? Do I have to create my own package? 

Br,
Lennart

**Comment by resyfer** [comment]
Hi,

You will find `pgexporter` in the [Common Yum Repository](https://yum.postgresql.org/packages/).

What the tutorial refers to, is adding the YUM repo. The simplest way is to execute the first line of the download instructions as per [this page](https://www.postgresql.org/download/) which has OS and version specific instructions.

For example, I'm on Fedora 38, so I will select `Linux` -> `Red Hat/Rocky/CentOS` -> `Version 16`, `Fedora 38`, `x86_64` and then execute the first line that says:
```sh
sudo dnf install -y https://download.postgresql.org/pub/repos/yum/reporpms/F-38-x86_64/pgdg-fedora-repo-latest.noarch.rpm
```

and then you can search (or install) the various packages:
```
$ sudo dnf search pgexporter
Last metadata expiration check: 0:05:44 ago on Fri 22 Sep 2023 12:11:39 PM IST.
============================================================================= Name Exactly Matched: pgexporter ==============================================================================
pgexporter.x86_64 : Prometheus exporter for PostgreSQL
============================================================================ Name & Summary Matched: pgexporter =============================================================================
pgexporter-debuginfo.x86_64 : Debug information for package pgexporter
pgexporter-debugsource.x86_64 : Debug sources for package pgexporter
pgexporter_ext_11.x86_64 : PostgreSQL extension to provide additional Prometheus metrics for pgexporter.
pgexporter_ext_12.x86_64 : PostgreSQL extension to provide additional Prometheus metrics for pgexporter.
pgexporter_ext_13.x86_64 : PostgreSQL extension to provide additional Prometheus metrics for pgexporter.
pgexporter_ext_14.x86_64 : PostgreSQL extension to provide additional Prometheus metrics for pgexporter.
pgexporter_ext_15.x86_64 : PostgreSQL extension to provide additional Prometheus metrics for pgexporter.
pgexporter_ext_16.x86_64 : PostgreSQL extension to provide additional Prometheus metrics for pgexporter.
```

I hope you got what you're looking for.

PS: There's an updated documentation under works, and it clears up this, and some other, confusion bits as well!

**Comment by caleno** [comment]
For reference 

I solve this by adding the Postgresql common repo to yum/dfn

_/etc/yum.repos.d/pgdg_common.repo_
```
[pgdg_common]
name=PostgreSQL common for RHEL / Rocky $releasever - $basearch
baseurl=https://download.postgresql.org/pub/repos/yum/common/redhat/rhel-$releasever-$basearch
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG
```

Thanks again for helping out with this :+1:
